### PR TITLE
review: feat: Add support to Java 10 'var' keyword

### PR DESCRIPTION
**Pull request by surli**
This PR add the support to Java 10 `var` keyword for local type inference.

### CODE REVIEW

File: src/main/java/spoon/reflect/path/CtRole.java [hunk 1]
```diff
@@ -101,7 +101,8 @@
 	OPENED_PACKAGE(MODULE_DIRECTIVE),
 	SERVICE_TYPE(MODULE_DIRECTIVE),
 	IMPLEMENTATION_TYPE,
-	PROVIDED_SERVICE(MODULE_DIRECTIVE);
+	PROVIDED_SERVICE(MODULE_DIRECTIVE),
+	INFERRED_VARIABLE;
```
Comment (monperrus) [inline src/main/java/spoon/reflect/path/CtRole.java]: I propose to name it IS_INFERRED

File: src/test/java/spoon/reflect/ast/AstCheckerTest.java [hunk 2]
```diff
@@ -158,6 +161,18 @@ private boolean isNotCandidate(CtMethod<?> candidate) {
 					|| notCandidates.contains(candidate.getDeclaringType().getSimpleName() + "#" + candidate.getSimpleName());
 		}
 
+		private boolean hasPrimitiveTypeForParameters(List<CtParameter<?>> parameters) {
```
Comment (monperrus) [inline src/test/java/spoon/reflect/ast/AstCheckerTest.java]: I don't see the relation with this PR. Why was it working before?

File: src/test/java/spoon/test/comment/CommentTest.java [hunk 3]
```diff
@@ -749,6 +749,7 @@ public void testDocumentationContract() throws Exception {
 		final Launcher launcher = new Launcher();
 		launcher.getEnvironment().setNoClasspath(true);
 		launcher.getEnvironment().setCommentEnabled(true);
+		launcher.getEnvironment().setComplianceLevel(10);
```
Comment (monperrus) [inline src/test/java/spoon/test/comment/CommentTest.java]: why is this required here?

File: src/test/java/spoon/reflect/ast/AstCheckerTest.java (hunk 2 above)
Comment (surli) [reply src/test/java/spoon/reflect/ast/AstCheckerTest.java]: I'm not sure. I suspect that on almost all other setters we were not using primitive types. I can extract it in another PR if you want: the relation here is that it fix a broken test resulting by the new setter.

File: src/test/java/spoon/test/comment/CommentTest.java (hunk 3 above)
Comment (surli) [reply src/test/java/spoon/test/comment/CommentTest.java]: I added a new snippet in the code with the following code:
```
var x = 10;
```

This won't compile if I do not use a compliance level to 10.

File: src/test/java/spoon/reflect/ast/AstCheckerTest.java (hunk 2 above)
Comment (monperrus) [reply src/test/java/spoon/reflect/ast/AstCheckerTest.java]: Bizarre. We have plenty of other setters with boolean (eg setImplicit, setShadow, etc). It would be much better to understand what happens here.

Comment (surli) [reply src/test/java/spoon/reflect/ast/AstCheckerTest.java]: I'll have a look

Comment (surli) [reply src/test/java/spoon/reflect/ast/AstCheckerTest.java]: Good thing you insisted on that one: I forgot to call the model change listener... I updated the contract here because it was not clear enough.

**Comment by monperrus** [comment]
That's great, thanks Simon!
